src/lib/operators/abstract_operator.cpp

```diff
@@ -96,6 +96,23 @@ void AbstractOperator::execute() {
         }
       }
     }
+    // Verify that nullability of columns and segments match for value segments
+    if (_output) {
+      for (auto chunk_id = ChunkID{0}; chunk_id < _output->chunk_count(); ++chunk_id) {
+        for (auto column_id = ColumnID{0}; column_id < _output->column_count(); ++column_id) {
+          const auto& abstract_segment = _output->get_chunk(chunk_id)->get_segment(column_id);
+          resolve_data_and_segment_type(*abstract_segment, [&](const auto data_type_t, const auto& segment) {
+            using ColumnDataType = typename decltype(data_type_t)::type;
+            using SegmentType = std::decay_t<decltype(segment)>;
+            if constexpr (std::is_same_v<SegmentType, ValueSegment<ColumnDataType>>) {
+              // If segment is nullable, the column must be nullable as well
+              Assert(!segment.is_nullable() || _output->column_is_nullable(column_id),
+                     "Nullable segment found in non-nullable column");
+            }
+          });
+        }
+      }
+    }
```

Comment:
Do I see it correctly that this loop iterating over all chunks and all columns is executed for every executed operator that has an output?

Reply:
Correct. In debug mode.

Reply:
Makes sense, didn't see it, never mind.